internal/api/integrations/k8s/k8s_test.go

```diff
@@ -0,0 +1,87 @@
+package k8s
+
+import (
+	"testing"
+
+	"ship-it/internal"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestAnnotationFor(t *testing.T) {
+	tests := []struct {
+		input    string
+		expected string
+	}{
+		{"word-counts", "helmreleases.k8s.wattpad.com/word-counts"},
+		{"", "helmreleases.k8s.wattpad.com/"},
+	}
+	for _, test := range tests {
+		assert.Equal(t, test.expected, annotationFor(test.input))
+	}
+}
+
+func TestGetImageForRepo(t *testing.T) {
+	tests := []struct {
+		inputMap map[string]interface{}
+		repo     string
+		expected *internal.Image
+	}{
+		{
+			map[string]interface{}{
+				"apples": "delicious",
+				"oranges": map[string]interface{}{
+					"taste": "delicious",
+				},
+				"image": map[string]interface{}{
+					"repository": "foo/bar",
+					"tag":        "baz",
+				},
+			},
+			"bar",
+			&internal.Image{
+				Registry:   "foo",
+				Repository: "bar",
+				Tag:        "baz",
+			},
+		}, {
+			map[string]interface{}{
+				"apples": "delicious",
+				"oranges": map[string]interface{}{
+					"taste": "delicious",
+				},
+				"image": map[string]interface{}{
+					"repository": "foo/oof",
+					"tag":        "baz",
+				},
+			},
+			"bar",
+			nil,
+		},
+	}
+
+	for _, test := range tests {
+		actualImg, _ := GetImageForRepo(test.repo, test.inputMap)
+		assert.Equal(t, test.expected, actualImg)
+	}
+}
+
+func TestFindRepo(t *testing.T) {
+	tests := []struct {
+		input    string
+		expected string
+	}{
+		{"https://github.com/Wattpad/highlander/tree/master/wattpad/src/services/word-counts", "highlander"},
+		{"https://github.com/Wattpad/miranda/", "miranda"},
+		{"https://github.com/Wattpad/", ""},
+		{"https://github.com/Wattpad", ""},
+	}
+	for _, test := range tests {
+		repo, err := findVCSRepo(test.input)
+		if err == nil {
+			assert.Equal(t, test.expected, *repo)
+		} else {
+			assert.Error(t, err)
```

Comment:
If I do it this way, test cases that have the function returning an error will fail. Since an error is always returned with an empty string, I will do `assert.Equal(t, test.expected, repo)` where the expected string is also empty.